lib/active_admin/async_exporter/reports/dsl.rb

```diff
@@ -14,10 +14,16 @@ def csv_async(decorate_model: false)
           csv_report(columns: csv_fields, decorate_model: decorate_model)
         end
 
-        def csv_column(column_name, column_value = nil)
+        def csv_column(column_name, column_value = nil, &block)
           column_value ||= column_name
 
-          csv_fields[:"#{column_name.to_sym}"] = column_value.to_sym
+          if block_given?
+            # For CR: Raise error if block.params.blank?
+            is_block = true
+            column_value = block.source
+          end
+
+          csv_fields[column_name.to_sym] = { block: is_block, value: column_value.to_s }
```

Comment:
```suggestion
           column_value = block.source if block_given?

          csv_fields[column_name.to_sym] = { block: block_given?, value: column_value.to_s }
```

I think this version would be simpler